src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,168 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Entry<T> head;
+    private Entry<T> tail;
+    private int size;
+
+    public MyLinkedList() {
+        head = null;
+        tail = null;
+        size = 0;
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        Entry<T> currentEntry = new Entry<T>(value, null, null);
+
+        if (size == 0) {
+            head = currentEntry;
+            tail = currentEntry;
+        } else {
+            Entry<T> prevEntry = tail;
+            prevEntry.next = currentEntry;
+
+            currentEntry.prev = prevEntry;
+            tail = currentEntry;
+        }
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+        } else if (index == 0) {
+            Entry<T> insertEntry = new Entry<T>(value, head, null);
+            head = insertEntry;
+            size++;
+        } else if (isValidIndex(index)) {
+            Entry<T> insertEntry = new Entry<T>(value, null, null);
 
+            Entry<T> currentWithIndex = getEntryByIndex(index);
+            if (currentWithIndex != null) {
+                currentWithIndex.prev = insertEntry;
+            }
+
+            Entry<T> previousEntry = getEntryByIndex(index - 1);
```

Comment:
do we need to call mathod  getEntryByIndex() twice?

Reply:
Updated. Done.